akka-actor/src/main/scala/akka/io/dns/internal/AsyncDnsResolver.scala

```diff
@@ -66,6 +68,18 @@ private[io] final class AsyncDnsResolver(
     }
   }
 
+  private def alreadyResolvedIp(knownToBeIp: String): Future[DnsProtocol.Resolved] = {
+    if (isInetAddress(knownToBeIp))
```

Comment:
Technically I'm checking this here for a second time but did not want to remove the check I guess

Reply:
I guess `isInetAdress` as implemented with this huge Regex is quite expensive so maybe not do it more than necessary.

Reply:
It is though we wanted to optimize it anyway, so figured to not spend time on that but fix the bugs for now

Reply:
Other proposal to restructure the code maybe? somewhat worried about removing the check from the method

Reply:
The name is clear and it is a private method. We could remove the check from the match and do the if inside

Reply:
ok removed